src/main/java/Parser.java

```diff
@@ -0,0 +1,122 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+// DEALS WITH MAKING SENSE OF USER'S COMMAND
+public class Parser {
+
+    protected Storage storage;
+    protected Ui ui;
+    protected TaskList tasks;
+
+    public Parser(Storage storage, Ui ui, TaskList tasks) {
+        this.storage = storage;
+        this.ui = ui;
+        this.tasks = tasks;
+    }
+
+    static String parseCommandType(String command) {
+        String[] tokenizedCommand = command.split(" ");
+        return tokenizedCommand[0];
+    }
+
+    static int parseTaskIndex(String command) {
+        String[] tokenizedCommand = command.split(" ");
+        return Integer.parseInt(tokenizedCommand[1]);
+    }
+
+    static String parseTodoDescription(String command) {
+        return command.substring(5);
+    }
+
+    static String parseDeadlineDescription(String command) {
+        return command.substring(9).split(" /by ")[0];
+    }
+
+    static LocalDate parseDeadlineDate(String command) {
+        return LocalDate.parse(command.substring(9).split(" /by ")[1]);
+    }
+
+    static String parseEventDescription(String command) {
+        return command.substring(6).split(" /at ")[0];
+    }
+
+    static String parseEventDate(String command) {
+        return command.substring(6).split(" /at ")[1];
+    }
+
+    static String parseKeyword(String command) {
+        return command.substring(5);
+    }
+
+    static boolean hasSaved(String command) {
+        return command.equals("save");
+    }
+
+    static boolean hasExited(String command) {
+        return command.equals("exit"); }
+
+    public String parseCommand(String command) {
```

Comment:
could try splitting this up into shorter parts so that the code isnt too nested with too many try catch blocks